dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java

```diff
@@ -1686,6 +1692,34 @@ public void testExtractWeekFromDateKeyword() {
         assertEquals(4000921042220002L, results.get(0).number);
     }
 
+    /**
+     * Verify that fetch type eager and lazy both work when using a detached entity returned by Jakarta Data
+     */
+    @Test
+    public void testFetchType() {
+        mobilePhones.removeAll();
+
+        List<String> apps = Arrays.asList("Settings", "Camera", "Phone", "Email", "Messages",
+                                          "UnoLingo", "BankApp", "SoloGame", "LocalNews");
+
+        List<String> emails = Arrays.asList("john.smith@example.com", "JohnDSmith@example.work.com");
+
+        // Populate database
+        UUID id = mobilePhones.insert(Mobile.of(OS.ANDROID, apps, emails)).deviceId;
+
+        // Outside of transaction, returned entity should be detached
+        Mobile johnsMobile = mobilePhones.findById(id).orElseThrow();
+
+        // Fetch type lazy should be populated when accessing apps field
+        assertFalse("Expected apps to be populated when using fetch type lazy", johnsMobile.apps.isEmpty());
+        assertTrue("Entity apps did not match expected apps", johnsMobile.apps.containsAll(apps));
+
+        // Fetch type eager emails field should be pre-populated
+        assertFalse("Expected emails to be populated when using fetch type eager", johnsMobile.emails.isEmpty());
+        assertTrue("Entity emails did not match expected apps", johnsMobile.emails.containsAll(emails));
```

Comment:
same reason as prior comment
```suggestion
        assertEquals("Entity apps did not match expected apps", apps, johnsMobile.apps);

        // Fetch type eager emails field should be pre-populated
        assertFalse("Expected emails to be populated when using fetch type eager", johnsMobile.emails.isEmpty());
        assertEquals("Entity emails did not match expected apps", emails, johnsMobile.emails);
```